backend/lucene/src/main/java/org/hibernate/search/backend/lucene/index/impl/IndexingBackendContext.java

```diff
@@ -66,4 +67,8 @@ Directory createDirectory(String indexName) throws IOException {
 		return new LuceneIndexWorkExecutor( workFactory, multiTenancyStrategy, orchestrator,
 				indexName, sessionContext );
 	}
+
+	public CompletableFuture<?> optimize(LuceneIndexWorkOrchestrator streamOrchestrator, String indexName) {
```

Comment:
Ok, the fact you have to add delegate methods all over the place is concerning. There may be something wrong with what we planned.

Don't you think it would be better if renamed "IndexWorkExecutor" to "IndexDocumentWorkExecutor", and moved the optimize/purge/flush operations to a new "IndexWorkExecutor" interface? Then you could remove the various operations you added in `MappedIndexManager` (just leave a createIndexWorkExecutor method) and here (same, just leave a createIndexWorkExecutor method).

Reply:
Good idea, I'll do it in a subsequent commit, since it requires a little bit of redesign.

Reply:
Sure this can definitely wait until you're done with everything else.

Reply:
What do you think of either `IndexBatchWorkExecutor` or `IndexMassiveWorkExecutor` or `IndexBulkWorkExecutor`  as a name for the new `IndexWorkExecutor' interface?

Reply:
What's wrong with `IndexWorkExecutor`/`IndexDocumentWorkExecutor`? That would make it clear that the first creates works scoped to the index, the second creates works scoped to the document.

If that doesn't work for you, then let's see the other names...

`Bulk` won't do because it means something different in Elasticsearch.
`Batch` won't do either, since it's not about batching: on the contrary, it's about executing one big work atomically.
`Massive` would work, I suppose, but it focuses on the fact these works are resource-intensive, while what's important is their scope.

`IndexGeneralWorkExecutor`/`IndexDocumentWorkExecutor`, maybe? `General` feels clumsy too, but at least it focuses on the scope.

Reply:
OK